integration_test/retl_test/sut_test.go

```diff
@@ -0,0 +1,359 @@
+				"config":                  dst.Config(),
+				"enabled":                 true,
+				"destinationDefinitionId": "xxxyyyzzSOU9pLRavMf0GuVnWV3",
+				"createdBy":               "xxxyyyzzueyoBz4jb7bRdOzDxai",
+				"workspaceId":             s.WorkspaceID,
+				"createdAt":               "2021-08-27T06:33:00.305Z",
+				"updatedAt":               "2021-08-27T06:33:00.305Z",
+				"isConnectionEnabled":     true,
+				"isProcessorEnabled":      true,
+				"destinationDefinition": map[string]any{
+					"config": map[string]any{
+						"transformAt":   "processor",
+						"transformAtV1": "processor",
+						"supportedMessageTypes": []string{
+							"alias",
+							"group",
+							"identify",
+							"page",
+							"screen",
+							"track",
+						},
+						"saveDestinationResponse": false,
+					},
+					"id":          "xxxyyyzzSOU9pLRavMf0GuVnWV3",
+					"name":        dst.TypeName(),
+					"displayName": strings.ToLower(dst.TypeName()),
+					"createdAt":   "2020-03-16T19:25:28.141Z",
+					"updatedAt":   "2021-08-26T07:06:01.445Z",
+				},
+			}
+		}
+		ss[i]["destinations"] = dd
+	}
+	c := map[string]any{
+		"enableMetrics": false,
+		"workspaceId":   s.WorkspaceID,
+		"sources":       ss,
+		"libraries":     []any{},
+	}
+	return c
+}
+
+func (s *SUT) Shutdown(t *testing.T) {
+	for _, srcWithDst := range s.Sources {
+		for _, dst := range srcWithDst.Destinations {
+			dst.Shutdown(t)
+		}
+	}
+
+	s.cancel()
+	<-s.done
+}
+
+// SendRETL sends a batch of records to the rETL endpoint.
+func (s *SUT) SendRETL(t *testing.T, sourceID, destinationID string, payload batch) {
+	t.Helper()
+	t.Logf("Sending rETL Events: %s -> %s", sourceID, destinationID)
+
+	var (
+		httpClient = &http.Client{}
+		method     = "POST"
+		url        = fmt.Sprintf("%s/internal/v1/retl", s.URL)
+	)
+
+	b, err := json.MarshalIndent(payload, "", "  ")
+	require.NoError(t, err)
+
+	t.Logf("sending records: %s", b)
+
+	req, err := http.NewRequest(method, url, bytes.NewReader(b))
+	if err != nil {
+		t.Logf("sendEvent error: %v", err)
+		return
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-Rudder-Source-Id", sourceID)
+	req.Header.Add("X-Rudder-Destination-Id", destinationID)
+
+	res, err := httpClient.Do(req)
+	if err != nil {
+		t.Logf("sendEvent error: %v", err)
+		return
+	}
+	defer func() { httputil.CloseResponse(res) }()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		t.Logf("sendEvent error: %v", err)
+		t.Fail()
+		return
+	}
+	if res.Status != "200 OK" {
+		t.Logf("sendEvent error: %v", string(body))
+		t.Fail()
+		return
+	}
+
+	t.Logf("Event Sent Successfully: (%s)", body)
+}
+
+// JobStatus hits the job-status endpoint and returns the status for a sourceID, jobRunID, jobTaskID.
+// If the job is not found, the second return value is false. Any other error is logged and the test is failed.
+func (s *SUT) JobStatus(t *testing.T, sourceID, jobRunID, jobTaskID string) (rsources.JobStatus, bool) {
```

Comment:
sourceID not being used?
```suggestion
func (s *SUT) JobStatus(t *testing.T, jobRunID, jobTaskID string) (rsources.JobStatus, bool) {
```